distributed/protocol/cupy.py

```diff
@@ -39,11 +33,18 @@ def __del__(self):
 @cuda_serialize.register(cupy.ndarray)
 def serialize_cupy_ndarray(x):
     # Making sure `x` is behaving
-    if not x.flags.c_contiguous:
+    if not (x.flags["C_CONTIGUOUS"] or x.flags["F_CONTIGUOUS"]):
```

Comment:
We could take this a step further and only make a copy if `x` is truly non-contiguous. I.e when checking contiguous, we can ignore all 0-strided (broadcasted), 1-sized dimensions, and the dimension order. 
We are properly not sending broadcasted arrays that often but this copy can makes it *very* expensive.

Reply:
Are you thinking of PR ( https://github.com/dask/distributed/pull/3474 )? Yeah it would be helpful to leverage a lot of the same tricks we are using with NumPy on other array-like serialization methods. Not entirely sure how we do that, but open to suggestions 🙂

F-order in particular is of interest for cuML, which largely uses F-order arrays. Hence the interest in handling that case zero-copy here.

cc @jrbourbeau (in case you have ideas 😉)